cylc/flow/network/graphql.py

```diff
@@ -66,10 +66,8 @@ def grow_tree(tree, path, leaves=None):
     """
     tree_loc = [tree, {}]
     for loc, key in enumerate(path):
-        if key in tree_loc[loc % 2]:
-            tree_loc[(loc + 1) % 2] = tree_loc[loc % 2][key]
-        else:
-            tree_loc[loc % 2][key] = tree_loc[(loc + 1) % 2]
+        tree_loc[(loc + 1) % 2] = tree_loc[loc % 2].get(
+            key, tree_loc[(loc + 1) % 2])
```

Comment:
Doesn't look like this does the same thing to me.

Reply:
I was just wondering the same.

Reply:
It isn't.

Reply:
Maybe keep the original and add a skip?

Reply:
Just pushed a fix up, alongside some tests for the unit in question.

Reply:
I've opened an issue on flake8-simply: https://github.com/MartinThoma/flake8-simplify/issues/83